chore(l2g): log annotated gold standards in w&b

ireneisdoomed opened the pull request:
## ✨ Context

This is the entry API to train the L2G model:
```
LocusToGeneTrainer.train(
                    gold_standard_data=data,
                    l2g_model=l2g_model,
                    model_path=model_path,
                    evaluate=True,
                    wandb_run_name=wandb_run_name,
                    **hyperparameters,
                )
```
Everything except the input `gold_standard_data` are just set parameters.
If we wanted to speed up experimenting as I suggest in https://github.com/opentargets/issues/issues/3253, we would just need to store `gold_standard_data`.

**We can use W&B for that**. We currently upload to W&B the train set of the annotated gold standards. I want to log the full matrix before splitting into train, test. Since we use a seed to split the data, this still makes sure we can reproduce the pipeline.

I like this idea because it binds the input data to the output model.

## 🛠 What does this PR implement
Changes in `LocusToGeneModel.evaluate` to accomplish the suggestion above.

## 🙈 Missing


## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [ ] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

DSuveges reviewed (approved):
It look OK. No obvious issues. Have you tried to run it?

ireneisdoomed commented:
It's running right now. It shouldn't have issues because we were logging a very similar table (only the train set). I just changed the input to the logging function and refactored the `train` function
